src/test/rpc/LedgerRPC_test.cpp

```diff
@@ -0,0 +1,283 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/protocol/ErrorCodes.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/test/jtx.h>
+#include <ripple/beast/unit_test.h>
+
+namespace ripple {
+
+class LedgerRPC_test : public beast::unit_test::suite
+{
+    static
+    std::unique_ptr<Config>
+    makeNonAdminConfig()
+    {
+        auto p = std::make_unique<Config>();
+        test::setupConfigForUnitTests(*p);
+        (*p)["port_rpc"].set("admin","");
+        (*p)["port_ws"].set("admin","");
+        return p;
+    }
+
+public:
+    void testBadInput()
+    {
+        testcase("Bad Input");
+        using namespace test::jtx;
+        Env env { *this };
+        Account const gw { "gateway" };
+        auto const USD = gw["USD"];
+        Account const bob { "bob" };
+
+        env.fund(XRP(10000), gw, bob);
+        env.trust(USD(1000), bob);
+        env.close();
+
+        {
+            Json::Value jvParams;
+            jvParams[jss::ledger_index] = "0"; // NOT an integer
+            auto const jrr = env.rpc ( "json", "ledger", to_string(jvParams) ) [jss::result];
+            BEAST_EXPECT(jrr[jss::error] == "invalidParams");
+            BEAST_EXPECT(jrr[jss::status] == "error");
+            BEAST_EXPECT(jrr[jss::error_message] == "ledgerIndexMalformed");
```

Comment:
> Your proposal might make it clearer whether we fail because of missing field or a value mismatch

Yeah, that's the idea. If it fails because the field is missing, you know you have a bigger problem than if it fails because the value got changed and the test "just" needs to be updated.